libraries/AP_BattMonitor/AP_BattMonitor_Params.cpp

```diff
@@ -176,6 +176,21 @@ const AP_Param::GroupInfo AP_BattMonitor_Params::var_info[] = {
     // @User: Advanced
     AP_GROUPINFO("OPTIONS", 21, AP_BattMonitor_Params, _options, 0),
 
+    // @Param: CELL_DET_V
+    // @DisplayName: Maximum cell voltage for cell count detection
+    // @Description: 
+    // @Units: V
+    // @Increment: 0.01
+    // @User: Advanced
+    AP_GROUPINFO("CELL_DET_V", 22, AP_BattMonitor_Params, _cell_detect_volt, 4.25f),
+
+    // @Param: CELL_COUNT
+    // @DisplayName: Overrides cell count autodetection if not 0
+    // @Description: 
+    // @Increment: 1
+    // @User: Advanced
+    AP_GROUPINFO("CELL_COUNT", 23, AP_BattMonitor_Params, _cell_count, 1),
```

Comment:
defaulting to 1 cell doesn't seem right

Reply:
Would you prefer to default to "autodetect", or something else?

Reply:
My objections to this panel were that a user just enabling it would have a high likelihood of getting incorrect information. I think that there is a way to avoid this and remove my primary objections:
1. Default the num cells param to -1, and display "---" for the value, like when info is missing...forcing the user to change the param
2. num cells = 0 is LIPO only autodetect. I think its a waste to try to add a param for cell max since the detect algorithm would probably be different for newer different chemistries anyway. Description of num of cells param should have clear warnings, for use with well charged LIPOs only.
3. >0 would be manual cell count.

this would force a user to make a conscious decision to accept the autodetect limitations or manually enter cell count before use.

Reply:
That seems reasonable. Amended to have the cell_count param to -1 by default and display `---` when this is the case